### PR TITLE
Report in status description when a repository configuration references invalid team handles

### PR DESCRIPTION
**Pull request by domtes-form3**
When the configuration references invalid team handles (teams are resolved either by `name` or by `slug`) the `github-team-approver` handler fails with a 500 error and the error message is only visible in logs.

With this change, when a PR is opened against a repository with a configuration referencing invalid team handles, the PR status is marked as `pending` and the description set to a readable message.

### CODE REVIEW

File: internal/api/stages/fakegithub/events.go
Comment (domtes-form3) [inline internal/api/stages/fakegithub/events.go:14]: Since both `PullRequestEvent` and `PullRequestReviewEvent` share the same data attributes, I renamed the type to `Event` and updated it with two specific constructors.

File: internal/api/approval/state.go
Comment (kdrabek-form3) [inline internal/api/approval/state.go:102]: What happens when both the slice of invalid team and pending team names are not empty? Would GTA still report that there are missing approvals as opposed to that the config is incorrect?

File: internal/api/approval/approval.go
Comment (wojciecholszewski-form3) [inline internal/api/approval/approval.go:173]: Wouldn't it be more obvious and explicit to test against certain error type prior assuming team handle is invalid?
```go
if err != nil {
  if errors.Is(ErrInvalidTeamHandle, err) {
    state.addInvalidTeamHandle(handle)
    continue
  }

  return err
}
```
Just to be 100% sure we don't dismiss any error we don't expect to be dismissed.

File: internal/api/approval/state.go
Comment (wojciecholszewski-form3) [reply internal/api/approval/state.go:102]: Yep, worth ensuring we don't overwrite result when latter case takes precedence over the former. Unless we can easily combine many error case into single string, which may be out of the scope of current improvement.

**Review by wojciecholszewski-form3 — COMMENTED** [review]
Minor remarks. Overall looks good. Happy to approve once my remarks are addressed/replied.

File: internal/api/approval/state.go
Comment (kdrabek-form3) [reply internal/api/approval/state.go:102]: AFAIR there's a character limit for this field, therefore I would vote GTA reports the most important problems first. And in this case misconfiguration seems more important :)

File: internal/api/approval/approval.go
Comment (renato-araujo-form3) [reply internal/api/approval/approval.go:178]: @wojciecholszewski-form3 I am not sure if there is any other possible error to be handled here, if you check [getTeamNameFromTeamHandle](https://github.com/form3tech-oss/github-team-approver/blob/7e893f127c7dfa96124ad5035f9f4557d27f1eb2/internal/api/approval/approval.go#L353) you will see that it will throw an error if the condition is not satisfied and other than that just panic or fatal in the compilation itself. 

Not sure what is the logic behind of the exception but this would be fine either:

```go
if err != nil { 
    state.addInvalidTeamHandle(handle)
    continue
}
```
or 

```go
if err != nil { 
    state.addInvalidTeamHandle(handle)
    return nil, err
}
```

Or just create a new error type and return it [here](https://github.com/form3tech-oss/github-team-approver/blob/master/internal/api/approval/approval.go#L351)

File: internal/api/approval/state.go
Comment (bart-lisiecki-form3) [reply internal/api/approval/state.go:102]: @kdrabek-form3 I and @mikolaj-krzyzanowski-f3 refactored this part so that it now sets the status as `error` and skips requesting reviews from the team handles.

We're preparing a functionality that creates a comment with invalid team handles in a separate branch.

**Review by wojciecholszewski-form3 — APPROVED** [review]

**Pull request merged**
